src/clients/dotnet/TigerBeetle/NativeClient.cs

```diff
@@ -113,45 +109,36 @@ public void ReleasePacket(Packet packet)
     {
         unsafe
         {
-            // It is unexpected for the client to be disposed here
-            // Since we wait for all acquired packets to be submitted and returned before disposing.
-            AssertTrue(client != IntPtr.Zero, "Client is closed");
-            AssertTrue(packet.Pointer != null, "Null packet pointer");
-            tb_client_release_packet(client, packet.Pointer);
+            Marshal.FreeCoTaskMem((IntPtr)packet.Pointer);
         }
     }
 
     public void Submit(Packet packet)
     {
         unsafe
         {
-            // It is unexpected for the client to be disposed here
-            // Since we wait for all acquired packets to be submitted and returned before disposing.
-            AssertTrue(client != IntPtr.Zero, "Client is closed");
-            tb_client_submit(client, packet.Pointer);
+            if (client != IntPtr.Zero)
+            {
+                lock (this)
+                {
+                    if (client != IntPtr.Zero)
+                    {
+                        tb_client_submit(client, packet.Pointer);
+                        return;
+                    }
+                }
+            }
+
+            packet.Pointer->status = PacketStatus.ClientShutdown;
+            OnComplete(packet.Pointer, null, 0);
         }
     }
 
     public Packet AcquirePacket()
     {
         unsafe
         {
-            if (client == IntPtr.Zero) throw new ObjectDisposedException("Client is closed");
-
-            TBPacket* packet;
-            var status = tb_client_acquire_packet(client, &packet);
-            switch (status)
-            {
-                case PacketAcquireStatus.Ok:
-                    AssertTrue(packet != null);
-                    return new Packet(packet);
-                case PacketAcquireStatus.ConcurrencyMaxExceeded:
-                    throw new ConcurrencyExceededException();
-                case PacketAcquireStatus.Shutdown:
-                    throw new ObjectDisposedException("Client is closing");
-                default:
-                    throw new NotImplementedException();
-            }
+            return new Packet((TBPacket*)Marshal.AllocCoTaskMem(sizeof(TBPacket)));
```

Comment:
Read that they were similar, but anything better is welcome. Was initially planning to use GCHandle.Alloc() but it takes an object rather than the struct. Will try out the MemoryPool approach.

Reply:
Any way to instantiate a `MemoryOwner<T>` from a `T*` or `IntPtr`? Current hurdle for using `MemoryPool`

Reply:
I think you should keep the `IMemoryOwner` reference inside the `Request`, so you can call `Dispose()` on it when the callback arrives.

Reply:
Went with this approach for now.